addons/project/models/project_task_recurrence.py

```diff
@@ -1,53 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import _, api, fields, models
+from odoo import _, api, fields, models, Command
 from odoo.exceptions import ValidationError
 
-from calendar import monthrange
 from dateutil.relativedelta import relativedelta
-from dateutil.rrule import rrule, rruleset, DAILY, WEEKLY, MONTHLY, YEARLY, MO, TU, WE, TH, FR, SA, SU
-
-MONTHS = {
-    'january': 31,
-    'february': 28,
-    'march': 31,
-    'april': 30,
-    'may': 31,
-    'june': 30,
-    'july': 31,
-    'august': 31,
-    'september': 30,
-    'october': 31,
-    'november': 30,
-    'december': 31,
-}
-
-DAYS = {
-    'mon': MO,
-    'tue': TU,
-    'wed': WE,
-    'thu': TH,
-    'fri': FR,
-    'sat': SA,
-    'sun': SU,
-}
-
-WEEKS = {
-    'first': 1,
-    'second': 2,
-    'third': 3,
-    'last': 4,
-}
 
 class ProjectTaskRecurrence(models.Model):
     _name = 'project.task.recurrence'
     _description = 'Task Recurrence'
 
-    task_template_id = fields.Many2one('project.task')
```

Comment:
For upgrade team, this field is not removed in the migration script because this field has been introduced in 16.2 and not before. 🙂